src/cc65/scanner.c

```diff
@@ -575,16 +582,28 @@ static void NumericConst (void)
     /* Since we now know the correct base, convert the input into a number */
     SB_SetIndex (&Src, Index);
     IVal = 0;
+    Overflow = 0;
     while ((C = SB_Peek (&Src)) != '\0' && (Base <= 10 ? IsDigit (C) : IsXDigit (C))) {
         DigitVal = HexVal (C);
         if (DigitVal >= Base) {
             Error ("Invalid digit \"%c\" beyond radix %u constant", C, Base);
             SB_Clear (&Src);
             break;
         }
-        IVal = (IVal * Base) + DigitVal;
+        if (((scan_t)(IVal * Base) / Base) != IVal) {
```

Comment:
Yes, you're right that I can do that in ones step. I will add this optimization.

IVal is not unsigned long. IVal is uint32_t and the specific size of unsigned long is already different on different platform targets, which is exactly the problem I'm trying to remedy with this cast. I'm not certain whether "unsigned" is a consistent size on all our platform targets either.

Reply:
The code shows that you don't need to resort to fixed sized data types and usage of "inttypes.h".

You can of course use `uint32_t` instead of "& 0xFFFFFFFF" but using the latter pattern is more generic and does also work in cases where there is no data type with a specific size. Say you have `long long` with at least 64 bits and want to check for 48 bit overflow. The method from my code works in this case, yours does not, because there is no uint48_t.

But as said above: This is only a suggestion and the final solution is completely up to you.